### PR TITLE
Added hints to three problems

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
could you follow the template i did in #3430 ?

File: solutions/gold/usaco-945.mdx [hunk 1]
```diff
@@ -25,6 +25,10 @@ This can be solved with dynamic programming. Keeping a 2d DP array, where $(j,i)
 
 </Spoiler>
 
+<Spoiler title="Solution">
+
+## Solution
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-945.mdx:30]: use ## Explanation instead of ## Solution

File: solutions/gold/usaco-863.mdx [hunk 2]
```diff
@@ -7,26 +7,29 @@ author: Sofia Yang, Ryan Chou
...
 
-Note on the constraints of the input, $\mathcal{O}(NK)$ is feasible.
+Notice the constraints on the input, O(NK) is feasible.
 
 </Spoiler>
 
 <Spoiler title="Hint 2">
 
-We notice that we just need to find the best possible sums for the first $i$ cows as we iterate $i$ from $1$ to $N$.
+We notice that we just need to find the best possible sums for the first i cows as we iterate i from 1 to N.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-863.mdx:19]: why remove the latex here

File: solutions/gold/usaco-945.mdx [hunk 3]
```diff
@@ -7,6 +7,27 @@ author: Sofia Yang, Ryan Chou
...
+
+<Spoiler title="Hint 2">
+
+We notice that we just need to find the minimum waste for the first $i$ snakes with only $j$ switches as we iterate $i$ and $j$ from $1$ to $N$ and $0$ to $K$.
+
+</Spoiler>
+
+<Spoiler title="Hint 3">
+
+This can be solved with dynamic programming. Keeping a 2d DP array, where $(j,i)$th element represents the minimum waste from $1$ to $i$th snake at $j$ switches.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-945.mdx:24]: this just gives the solution lol
i'd remove it at this point

File: solutions/gold/usaco-863.mdx [hunk 4]
```diff
@@ -7,7 +7,28 @@ author: Sofia Yang, Ryan Chou
...
+
+<Spoiler title="Hint 2">
+
+We notice that we just need to find the best possible sums for the first i cows as we iterate i from 1 to N.
+
+</Spoiler>
+
+<Spoiler title="Hint 3">
+
+This can be solved with dynamic programming. Keeping a DP array, where ith element represent the best possible sums from 1 to ith cow.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-863.mdx:25]: same here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/usaco-623.mdx [hunk 5]
```diff
@@ -7,6 +7,26 @@ author: Jeffrey Zhang, Ryan Chou
...
+
+<Spoiler title="Hint 1">
+
+2D grid representations can often be unwieldy. What if we thought of the problem in terms of a graph?
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+We can turn each $(n+1)(m+1)$ into a node, and the edge weight between the four surrounding nodes their shared side length.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-623.mdx:19]: each $(n+1)(m+1)$ what into a node?

File: solutions/gold/usaco-945.mdx [hunk 6]
```diff
@@ -7,6 +7,21 @@ author: Sofia Yang, Ryan Chou
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
 
+<Spoiler title="Hint 1">
+
+Note on the constraints of the input, setting $K$ upperbound as $N$, $\mathcal{O}(N^3)$ is feasible.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-945.mdx:12]: > setting $K$ upperbound as $N$

what this mean

**Comment by SansPapyrus683** [comment]
@envyaims @ryanchou-dev

File: solutions/gold/usaco-863.mdx [hunk 7]
```diff
@@ -7,6 +7,21 @@ author: Sofia Yang, Ryan Chou
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_teamwork_gold_dec18.html)
 
+
+<Spoiler title="Hint 1">
```
Comment (envyaims) [inline solutions/gold/usaco-863.mdx:11]: im ngl these are barely hints

File: solutions/gold/usaco-863.mdx [hunk 8]
```diff
@@ -7,6 +7,21 @@ author: Sofia Yang, Ryan Chou
...
+
+Notice the constraints on the input, $\mathcal{O}(N \cdot K)$ is feasible.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+We notice that we just need to find the best possible sums for the first $i$ cows as we iterate $i$ from $1$ to $N$.
```
Comment (envyaims) [inline solutions/gold/usaco-863.mdx:19]: is this supposed to be a hint at dp? im ngl for more inexperienced users this hint probably means nothing

File: solutions/gold/usaco-863.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-863.mdx:11]: in terms of "no crap" or "gives you the solution"

File: solutions/gold/usaco-945.mdx [hunk 9]
```diff
@@ -15,7 +15,7 @@ Note on the constraints of the input, the maximum number of changes to the size
 
 <Spoiler title="Hint 2">
 
-We notice that we just need to find the minimum waste for the first $i$ snakes with only $j$ switches as we iterate $i$ and $j$ from $1$ to $N$ and $0$ to $K$.
+We notice that we just need to find the minimum waste for the first $i$ snakes with only $j$ switches as we iterate $i$ and $j$ from $1$ to $N$ and $0$ to $K$. Notice this nature of computing solutions fits with the idea of dynamic programming.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-945.mdx:18]: i'd leave this as it was tbh

File: solutions/gold/usaco-863.mdx [hunk 10]
```diff
@@ -10,13 +10,13 @@ author: Sofia Yang, Ryan Chou
...
 
-Notice the constraints on the input, $\mathcal{O}(N \cdot K)$ is feasible.
+Notice the constraints on the input, $\mathcal{O}(N \cdot K)$ is feasible. Meaning, for each of the $N$ cows, we need to calculate something for them $K$ times.
 
 </Spoiler>
 
 <Spoiler title="Hint 2">
 
-We notice that we just need to find the best possible sums for the first $i$ cows as we iterate $i$ from $1$ to $N$.
+We notice that we just need to find the best possible sums for the first $i$ cows as we iterate $i$ from $1$ to $N$. Notice this nature of computing solutions fits with the idea of dynamic programming.
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-863.mdx:19]: same here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
after envy approves i'll merge

**Review by TruaShamu — APPROVED** [review]

File: solutions/gold/usaco-863.mdx (hunk 7 above)
Comment (envyaims) [reply solutions/gold/usaco-863.mdx:11]: no crap

Comment (envyaims) [reply solutions/gold/usaco-863.mdx:11]: 99.98% of users at this level would already think of NK and not some other weird obscure complexity.
```Meaning, for each of the $N$ cows, we need to calculate something for them $K$ times.```
Damn thats crazy, at least specify "something"

Comment (RedBlueBird) [reply solutions/gold/usaco-863.mdx:11]: How about now?

Comment (envyaims) [reply solutions/gold/usaco-863.mdx:11]: how about
`we can recalculate the most optimal sums of skill levels $K$ times`?

Comment (RedBlueBird) [reply solutions/gold/usaco-863.mdx:11]: > how about `we can recalculate the most optimal sums of skill levels $K$ times`?

oh ok, how about now

**Review by envyaims — APPROVED** [review]
:o

**Pull request merged**